runtime/src/main/java/org/corfudb/runtime/LogReplicationUtils.java

```diff
@@ -1,12 +1,179 @@
+
+    private LogReplicationUtils() { }
+
+    public static void subscribe(@Nonnull LogReplicationListener clientListener, @Nonnull String namespace,
+                          @Nonnull String streamTag, @Nonnull List<String> tablesOfInterest, int bufferSize,
+                          CorfuStore corfuStore) {
+
+        long subscriptionTimestamp = getSubscriptionTimestamp(corfuStore, namespace, clientListener);
+        corfuStore.getRuntime().getTableRegistry().getStreamingManager().subscribeLogReplicationListener(
+                clientListener, namespace, streamTag, tablesOfInterest, subscriptionTimestamp, bufferSize);
+        log.info("Client subscription at timestamp {} successful.", subscriptionTimestamp);
+    }
+
+
+    private static long getSubscriptionTimestamp(CorfuStore corfuStore, String namespace,
+                                                 LogReplicationListener clientListener) {
+        Optional<Counter> mvoTrimCounter = MicroMeterUtils.counter("logreplication.subscribe.trim.count");
+        Optional<Counter> conflictCounter = MicroMeterUtils.counter("logreplication.subscribe.conflict.count");
+        Optional<Timer.Sample> subscribeTimer = MicroMeterUtils.startTimer();
+
+        Table<LogReplicationSession, ReplicationStatus, Message> replicationStatusTable =
+            openReplicationStatusTable(corfuStore);
+
+        try {
+            return IRetry.build(IntervalRetry.class, () -> {
+                try (TxnContext txnContext = corfuStore.txn(namespace)) {
+                    // The transaction is started in the client's namespace and the Replication Status table resides in the
+                    // system namespace.  Corfu Store does not validate the cross-namespace access as long as there are no
+                    // writes on the table in the different namespace.  This hack is required here as we want client full
+                    // sync to happen in the same transaction which checks the status of a snapshot sync so that there is no
+                    // window between the check and full sync.
+                    List<CorfuStoreEntry<LogReplicationSession, ReplicationStatus, Message>> entries =
+                        txnContext.executeQuery(replicationStatusTable,
+                                record -> record.getKey().getSubscriber().getModel()
+                                    .equals(LogReplication.ReplicationModel.LOGICAL_GROUPS));
+
+                    // Sharding in Log Replication is done based on a destination.  So for a given replication model,
+                    // any Sink node will have a session with only 1 Source node.  Hence, there should be only 1
+                    // record corresponding to the Logical Group Replication Model.
+                    Preconditions.checkState(entries.size() == 1);
+                    CorfuStoreEntry<LogReplicationSession, ReplicationStatus, Message> entry = entries.get(0);
```

Comment:
Not sure if we discussed this before - should we consider the case where there is no status table entry set on this sink? (i.e. `entries.size() == 0`)    

I encountered this issue when try to do some basic verification in LogicalGroupIT, and it happens when:
- `subscirbeLRListener` before `startReplicationServers` so there is not entry
- `subscirbeLRListener` right after `startReplicationServers` and there is a race condition that status table on this sink could have not been set

will that happen in real deployment?

Reply:
yes it's a valid point.  It can happen on a fresh deployment when this session has either not yet been discovered or not yet initialized.  Adding a check to accommodate it.  Thanks.